### ISSUE TITLE
KeyError: Package('pycurl', '7.44.1', source_type='legacy' ...) on celery update

### ISSUE THREAD

**Issue by HoneyryderChuck** [python-poetry/poetry#5019]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: docker image (`python:3.9-buster`)
- **Poetry version**: 1.1.12

```toml
# relevant contents of the toml
boto3 = "*"
celery = {extras = ["sqs"], version = "*"}
# this is what I have to have in order for things to work
# pycurl = "*"
```

## Issue

I want to update `celery` to the latest version (5.2.3). Nothing in the `pyproject.toml` stops me, i.e. no transitive dependency mismatches. However, when I `poetry update`, it's unable to upgrade `pycurl` to latest deployed version. So then I add `pycurl` to the toml, although I don't use it directly. update works. Now, I have to keep `pycurl` around in the toml for everything to work. If I remove it, I get the following error on `poetry install`:

```
KeyError

  Package('pycurl', '7.44.1', source_type='legacy', source_url='https://our-pypi-proxy', source_reference='company')

  at ~/.local/lib/python3.9/site-packages/poetry/puzzle/solver.py:270 in _solve
      266│
      267│                 continue
      268│
      269│             final_packages.append(package)
    → 270│             depths.append(results[package])
      271│
      272│         # Return the packages in their original order with associated depths
      273│         return final_packages, depths
      274│
```

I think I should have to carry a transitive dependency in the toml. Can someone see what's wrong?

**Comment by HoneyryderChuck** [comment]
Indeed, that's the workaround.  But it shouldn't be required, that's poetry job.

**Comment by henrysingleton** [comment]
Also happening for me, identical behaviour for poetry versions 1.1.4, 1.1.11 and 1.1.12. 
Using Python 3.9.7. 

Example dependencies that will reproduce the issue: 

```
[tool.poetry.dependencies]
python = "^3.9"
celery = {extras = ["sqs"], version = "5.2.3"}
```

Either manually installing pycurl first or adding `pycurl = "*"` to requirements also worked for me.

**Comment by dimbleby** [comment]
duplicate #4683

**Comment by philipGaudreau** [comment]
For me, the solution was to change my python's version. Went from 3.9.X to 3.8.X. Not in my pyproject's file, but my global one.

**Comment by travellingprog** [comment]
Looks like this will be fixed in an upcoming Poetry 1.2.x release. In the meanwhile, I also installed boto3 and pycurl manually, before installing celery[sqs]. For Celery v5.2.3, the expected version range for each dependency can be found here: https://github.com/celery/kombu/blob/v5.2.3/requirements/extras/sqs.txt
